.github/workflows/lint.yaml

```diff
@@ -0,0 +1,39 @@
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+name: Lint
+
+jobs:
+  lint:
+    name: mypy
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Setup micromamba
+        uses: mamba-org/provision-with-micromamba@v15
+        with:
+          environment-file: ci/environment.yml
+          environment-name: spherely-dev
+          extra-specs: |
+            python=3.11
+
+      - name: Build and install spherely
+        run: |
+          python -m pip install . -v --no-build-isolation
+
+      - name: Install mypy
+        run: |
+          python -m pip install 'mypy<0.990'
+
+      - name: Run mypy
+        run: |
+          python -m mypy --install-types --non-interactive
```

Comment:
Shall we also run pre-commit in this lint build?

Reply:
Like https://github.com/geopandas/geopandas/blob/571e0b15f61dfbf957e347d166118444da712afd/.github/workflows/tests.yaml#L22 or https://github.com/shapely/shapely/blob/bc62f989c8bb2de38f9762a301b7b02c416d38b9/.github/workflows/lint.yml#L16-L17

Reply:
I installed pre-commit as a github app for this repository.

Reply:
Ah, yes, I see it now in the "checks" listing in the UI